async-inn/Controllers/HotelRoomsController.cs

```diff
@@ -91,6 +97,7 @@ public async Task<IActionResult> PutHotelRoom(int hotelId, int roomNumber, Hotel
 
         // POST: api/HotelRooms
         // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [Authorize(Roles = "District Manager, Property Manager")]

```

Comment:
The way I interpret the specs is that a Property Manager can add an amenity to a room (as in the route AddAmenityToRoom in your Rooms Controller) But that ultimately only a District Manager can create a new amenity.